src/plugins/workspace/public/components/workspace_form/data_source_connection_table.tsx

```diff
@@ -284,48 +244,14 @@ export const DataSourceConnectionTable = ({
   };
 
   return (
-    <>
-      <EuiInMemoryTable
-        items={openSearchConnections}
-        itemId="id"
-        columns={columns}
-        selection={selection}
-        search={search}
-        key={connectionType}
-        isSelectable={true}
-        itemIdToExpandedRowMap={itemIdToExpandedRowMap}
-        isExpandable={true}
-        className="workspace-detail-direct-query-table"
-        pagination={{
-          initialPageSize: 10,
-          pageSizeOptions: [10, 20, 30],
-        }}
-      />
-      <EuiSpacer />
-      {modalVisible && (
-        <EuiConfirmModal
-          data-test-subj="workspaceForm-cancelModal"
-          title={i18n.translate('workspace.detail.dataSources.modal.title', {
-            defaultMessage: 'Remove data source(s)',
-          })}
-          onCancel={() => {
-            setModalVisible(false);
-            setSelectedItems([]);
-          }}
-          onConfirm={() => {
-            setModalVisible(false);
-            handleUnassignDataSources(selectedItems);
-          }}
-          cancelButtonText={i18n.translate('workspace.detail.dataSources.modal.cancelButton', {
-            defaultMessage: 'Cancel',
-          })}
-          confirmButtonText={i18n.translate('workspace.detail.dataSources.Modal.confirmButton', {
-            defaultMessage: 'Remove data source(s)',
-          })}
-          buttonColor="danger"
-          defaultFocusedButton="confirm"
-        />
-      )}
-    </>
+    <EuiInMemoryTable
+      {...(restProps as EuiInMemoryTableProps<DataSourceConnection>)}
```

Comment:
```suggestion
      {...restProps}
```
We'd better avoid type assertion.